### DISCUSSION TITLE
Teams Ai library, how to add Actions

### DISCUSSION THREAD

**Discussion by RollsChris** [microsoft/teams-sdk#1198] [Q&A]
Hi,

I have recently tried to add actions to my bot, 

i added this code to app.ai:

```javascript
app.ai.action("reset", async (context: TurnContext, state: ApplicationTurnState) => {
  state.deleteConversationState();
  await context.sendActivity(`Ok I've deleted the current conversation state.`);
  return AI.StopCommandName;
})
```
I have my actions.json in my prompts/chat folder 
```json
[
    {
        "name": "reset",
        "description": "Reset conversation"
    }
]
```

What else am i missing? where is the best explanation of this stuff?

Thanks

**Comment by corinagum** [comment]
Hi @RollsChris, do you also have your `skprompt.txt` in your prompts folder? 

We have plenty of documentation to get started! 
- For our developer-maintained docs, please take a look at the [getting-started folder](https://github.com/microsoft/teams-ai/tree/main/getting-started).
- For an introduction to Copilot, there is a [youtube video introduction](https://www.youtube.com/watch?v=A3_xnmIxv6c).
- Our official docs can be found at [Microsoft Learn](https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/teams%20conversational%20ai/teams-conversation-ai-overview).

**Reply by RollsChris** [reply]
Hi, my skprompts.txt looks like this: 

```
You are a sophisticated chatbot assistant.
Only use the functions you have been provided with.
Don't make assumptions about what values to use with functions.
Ask for clarification if a user request is ever ambiguous.
Keep your response short and concise.
```
i didnt think this had much effect on the actions as the sample project included only shows:

```
The following is a conversation with an AI assistant.
The assistant can turn a light on or off.

context:
The lights are currently {{getLightStatus}}.
```

thanks chris

**Reply by RollsChris** [reply]
I have been through the documentation and i think its lacking a good explanation of how actions work, for example i haven't seen anywhere that it talks about action.json file? but i can see that its used in the in the examples. 

I may be missing something but surely there should be a mention in the ACTIONS.md in the concepts folder?

also nothing shows up googling Teams ai actions.json

**Comment by RollsChris** [comment]
found a mention of action.json in the augmentation docs. Ill try and work through it tomorrow,  I can see a potential issue though, we are using a datasource augmentation, can you have more than one? so one for actions and datasource:

this is my config.json

```json
{
  "schema": 1,
  "description": "AI Bot",
  "type": "chat",
  "completion": {
    "max_tokens": 500,
    "temperature": 0.9,
    "top_p": 0.0,
    "presence_penalty": 0.6,
    "frequency_penalty": 0.0

  },"augmentation": {
    "augmentation_type": "none",
    "data_sources": {
        "teams-ai": 1200
    }
}
}
```

**Reply by corinagum** [reply]
Hi @RollsChris, sorry for the delayed check-in -- how did this go for you? 
`action.json` is unique per prompt, and every prompt can have augmentations / data-sources.

**Reply by lifan0127** [reply]
Hi @corinagum , how does the Action Planner choose from multiple prompts? In my case, the default prompt is always used. Thanks!

**Reply by lifan0127** [reply]
Sorry, I realized that I can specify the prompt in `planner.completePrompt`.
